addons/website_blog/models/website_blog.py

```diff
@@ -19,8 +19,23 @@ class Blog(osv.Model):
     _columns = {
         'name': fields.char('Blog Name', required=True, translate=True),
         'subtitle': fields.char('Blog Subtitle', translate=True),
+        'active': fields.boolean('Active'),
     }
 
+    _defaults = {
+        'active': True,
+    }
+
+    def write(self, cr, uid, ids, vals, context=None):
+        res = super(Blog, self).write(cr, uid, ids, vals, context=context)
+        if 'active' in vals:
+            # archiving/unarchiving a blog does it on its posts, too
+            BlogPost = self.pool['blog.post']
+            ctx = dict(context, active_test=False)
+            posts = BlogPost.search(cr, uid, [('blog_id', 'in', ids)], context=ctx)
+            BlogPost.browse(cr, uid, posts, context=context).write({'active': vals['active']})
```

Comment:
why need to browse then write? as is it old api write is not enough ? 
BlogPost.write(cr, uid, posts, {'active': vals['active']})